src/rgw/rgw_basic_types.h

```diff
@@ -594,6 +594,9 @@ inline std::ostream& operator<<(std::ostream& out, const rgw_user &u) {
   u.to_str(s);
   return out << s;
 }
-
-
+struct prefetch_range {
+    size_t off = 0;
+    uint64_t len = 0;
+    bool prefetch_data = false;
```

Comment:
this boolean was already implied by the presence or lack of a `prefix_range` pointer. but you changed `rgw_build_object_policies()` to take this argument by value rather than by pointer, making it required even if you don't want prefetching - how come?

nit: use two spaces for indent